chia/wallet/wallet_coin_store.py

```diff
@@ -62,58 +62,65 @@ async def create(cls, wrapper: DBWrapper):
         await self.db_connection.execute("CREATE INDEX IF NOT EXISTS wallet_id on coin_record(wallet_id)")
 
         await self.db_connection.commit()
-        self.coin_record_cache = dict()
-        self.coin_wallet_record_cache = {}
-        all_coins = await self.get_all_coins()
-        for coin_record in all_coins:
-            self.coin_record_cache[coin_record.coin.name()] = coin_record
-
-        self.wallet_cache_lock = asyncio.Lock()
+        self.coin_record_cache = {}
+        self.unspent_coin_wallet_cache = {}
+        await self.rebuild_wallet_cache()
         return self
 
     async def _clear_database(self):
         cursor = await self.db_connection.execute("DELETE FROM coin_record")
         await cursor.close()
         await self.db_connection.commit()
 
+    async def rebuild_wallet_cache(self):
+        # First update all coins that were reorged, then re-add coin_records
+        all_coins = await self.get_all_coins()
```

Comment:
this returns all coins in this wallet, including spent ones, right?
It's not clear that this will fit in memory, especially not when parsed into python types.

Reply:
This is only coins that are owned by this wallet, so its much less than the whole blockchain

Reply:
Yes, this is for single wallet. Not concerned about memory size here.